lang/cs/Org.Apache.REEF.Bridge/InteropUtil.cpp

```diff
@@ -129,6 +129,21 @@ JNIEnv* RetrieveEnv(JavaVM* jvm) {
   return env;
 }
 
+
+const int defaultRecursionDepthForExceptionFormat = 3;
+
 String^ FormatJavaExceptionMessage(String^ errorMessage, Exception^ exception) {
-  return String::Concat(errorMessage, Environment::NewLine, exception->StackTrace);
+	return FormatJavaExceptionMessage(errorMessage, exception, defaultRecursionDepthForExceptionFormat);
+}
+
+String^ FormatJavaExceptionMessage(String^ errorMessage, Exception^ exception, int recursionDepth) {
+	
+	return (!exception)
+		? String::Concat(errorMessage, "null")
+		: recursionDepth >= 0
+			? String::Concat(errorMessage, Environment::NewLine,
+				exception->Message, Environment::NewLine,
+				exception->StackTrace, Environment::NewLine,
+				FormatJavaExceptionMessage( "Nested Exception: ", exception->InnerException, --recursionDepth), Environment::NewLine)
```

Comment:
I think if there are several nested exceptions this will accumulate multiple `NewLine`s in the end of the message. In all other branches we don't end the message with new line so we probably should remove it from here as well.